core/src/main/java/com/datastax/oss/driver/internal/core/os/JnrNativeImpl.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.os;
+
+import com.datastax.oss.driver.internal.core.util.Reflection;
+import java.lang.reflect.Method;
+import java.util.Optional;
+import java.util.function.Supplier;
+import jnr.ffi.Platform;
+import jnr.posix.POSIX;
+import jnr.posix.POSIXFactory;
+import jnr.posix.Timeval;
+import jnr.posix.util.DefaultPOSIXHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JnrNativeImpl implements NativeImpl {
+
+  private static final Logger LOG = LoggerFactory.getLogger(JnrNativeImpl.class);
+
+  private final Optional<POSIX> posix;
+  private final Optional<Platform> platform;
```

Comment:
We allow users to [exclude JNR dependencies manually](https://docs.datastax.com/en/developer/java-driver/4.5/manual/core/integration/#native-libraries). If those JARs are missing from the classpath, `Native` should still load correctly, even though all `isXxxAvailable()` methods will return false.

This doesn't work with this version, if I exclude JNR I get:
```
java.lang.NoClassDefFoundError: jnr/posix/POSIXHandler
	at com.datastax.oss.driver.internal.core.os.Native.<clinit>(Native.java:23)
```

I suggest we simply copy over the code from 4.x, with the `PosixLoader` and `PlatformLoader` inner classes. It's a bit less pretty, but it's tried and tested.

Reply:
Ah, that explains the NoClassDefFoundError catch blocks in the old code.  I thought that was handling something else but now it makes considerably more sense.

There's no need to copy in the old loader code; all we really need here is a loader which can handle exceptions coming off attempts to create instances.  I'd argue that's actually a cleaner way to represent what's occurring in this case; the old way (loading everything fine and then having NCDFE checks around the "isXAvailable" calls) is far less clear (as evidenced by my confusion above).

I'll put together a quick change to handle this... it shouldn't be too bad.

Reply:
OK, I saw your empty implementation approach, that works for me.